pkg/codegen/go/gen_test.go

```diff
@@ -65,7 +68,51 @@ func TestGeneratePackage(t *testing.T) {
 	generatePackage := func(tool string, pkg *schema.Package, files map[string][]byte) (map[string][]byte, error) {
 		return GeneratePackage(tool, pkg)
 	}
-	test.TestSDKCodegen(t, "go", generatePackage)
+	test.TestSDKCodegen(t, "go", generatePackage, typeCheckGeneratedPackage)
+}
+
+func typeCheckGeneratedPackage(t *testing.T, pwd string) {
+	var err error
+	var ex string
+	ex, err = executable.FindExecutable("go")
+	require.NoError(t, err)
+
+	// go SDKs live in their own folder:
+	// typecheck/go/$NAME/$FILES
+	// where FILES contains all the project files (including go.mod)
+	//
+	// This is not true when `package.language.go.rootPackageName` is set.
+	dir, err := ioutil.ReadDir(pwd)
+	require.NoError(t, err)
+	root := pwd
+	if len(dir) == 1 {
+		require.True(t, dir[0].IsDir())
+		root = filepath.Join(pwd, dir[0].Name())
+	}
+	t.Logf("*** Testing go in dir: %q ***", root)
+
+	var stdout, stderr bytes.Buffer
+	cmdOptions := integration.ProgramTestOptions{Stderr: &stderr, Stdout: &stdout}
+	// We don't want to corrupt the global go.mod and go.sum with packages we
```

Comment:
Huh interesting, do I understand correctly that codegen does not emit a go.mod file? So we need to do it here? SGTM in principle.

Reply:
Codegen does not emit a `go.mod` file. We have a very strange build tree for go projects by default. I think these are related, but it's not clear. 
```
pulumi/sdk
├── dotnet
├── go
├── go.mod
├── go.sum
├── nodejs
├── proto
├── python
└── README.md
```